### ISSUE TITLE
Inconsistent results when run ansible-lint command and run ansible-lint with pre-commit

### ISSUE THREAD

**Issue by netsandbox** [ansible-community/ansible-lint#1757]
##### Summary
If you run `ansible-lint` in an Ansible project (without any positional arguments), yaml files are detected as **yaml** files.

If you run ansible-lint with `pre-commit`, as pre-commit send each changed yaml file as a positional argument to ansible-lint,
yaml files are detected as **playbook**.
This results from the changes in PR #1273:
https://github.com/ansible-community/ansible-lint/pull/1273/files#diff-d453523679145f511ca4e8fb97be96e7dd664ae88373e2e5e9a90a65fd751cabR760-R768

Sadly the PR and commit doesn't tell us why files provided as positional argument to ansible-lint are always treated as playbooks, maybe @ssbarnea can tell us here more.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
ansible --version
ansible [core 2.11.6] 
  config file = /home/user/git/ansible/ansible.cfg
  configured module search path = ['/home/user/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/user/git/ansible/venv/lib/python3.8/site-packages/ansible
  ansible collection location = /home/user/git/ansible/collections
  executable location = /home/user/git/ansible/venv/bin/ansible
  python version = 3.8.10 (default, Sep 28 2021, 16:10:42) [GCC 9.3.0]
  jinja version = 3.0.2
  libyaml = True

ansible-lint --version
ansible-lint 5.2.1 using ansible 2.11.6
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### OS / ENVIRONMENT

Ubuntu 20.04

##### STEPS TO REPRODUCE

* create a minimal ansible project
* add a test.yaml file with keys not recognized by ansible
* add a pre-comit config as described in https://ansible-lint.readthedocs.io/en/latest/configuring.html#pre-commit-setup
* run `ansible-lint` (test.yaml is detected as yaml file and shows no error)
* run `ansible-lint test.yaml ` (test.yaml is detected as playbook and shows an error)
* run `pre-commit run --files test.yaml ansible-lint` (test.yaml is detected as playbook and shows an error)

##### Desired Behaviour

Yaml files provided to ansible-lint as positional argument **should not** always be treated as playbooks.

##### Actual Behaviour
Yaml files provided to ansible-lint as positional argument are always be treated as playbooks.

**Comment by ssbarnea** [comment]
Not a bug, documented behavior. I am sure that the example for using internet with pre-commit does have pass_filenames disabled.

**Comment by ssbarnea** [comment]
Look at https://github.com/ansible-community/ansible-lint/blob/main/.pre-commit-hooks.yaml#L13 and https://ansible-lint.readthedocs.io/en/latest/configuring.html#pre-commit-setup

If you follow that guideline you will get the same kind of behavior for running ansible-lint without arguments when running with pre-commit. Ansible-lint cannot reliably be run against specific files because it needs context. For example a task file cannot be evaluated in absence of a playbook or role that includes it.
